src/components/nabla/Pools/Backstop/WithdrawLiquidity/useBackstopDrain.ts

```diff
@@ -0,0 +1,65 @@
+import { useCallback } from 'react';
+import { config } from '../../../../../config';
+import { backstopPoolAbi } from '../../../../../contracts/nabla/BackstopPool';
+import { subtractBigDecimalPercentage } from '../../../../../helpers/calc';
+import { getValidSlippage } from '../../../../../helpers/transaction';
+import { decimalToRaw } from '../../../../../shared/parseNumbers/metric';
+import { NablaInstanceBackstopPool, NablaInstanceSwapPool } from '../../../../../hooks/nabla/useNablaInstance';
+import { useContractWrite } from '../../../../../hooks/nabla/useContractWrite';
+import { WithdrawLiquidityValues } from './useWithdrawLiquidity';
+import { UseQuoteResult } from '../../../../../hooks/nabla/useQuote';
+
+export type UseBackstopDrainProps = {
+  backstopPool: NablaInstanceBackstopPool;
+  selectedSwapPool: NablaInstanceSwapPool | undefined;
```

Comment:
nit: `selectedSwapPool?:NablaInstanceSwapPool`

Reply:
Both types are almost the same, but you see that I do this style a lot if the value for that parameter is always provided at every call site (instead of being left out sometimes) – I can change if it clashes with the style in the rest of the codebase.